lib/location/location_core.c

```diff
@@ -428,6 +428,35 @@ void location_core_event_cb(const struct location_data *location)
 		}
 		LOG_DBG("  Google maps URL: https://maps.google.com/?q=%s,%s",
 			log_strdup(latitude_str), log_strdup(longitude_str));
+		if (current_config.mode == LOCATION_REQ_MODE_ALL) {
+			/* Get possible next method */
+			previous_method = current_event_data.location.method;
+			current_method_index++;
+			if (current_method_index < current_config.methods_count) {
+				requested_method =
+					current_config.methods[current_method_index].method;
+				LOG_INF("LOCATION_REQ_MODE_ALL: acquired location using '%s', "
+					"trying with '%s' next",
+					(char *)location_method_api_get(
+						previous_method)->method_string,
+					(char *)location_method_api_get(
+						requested_method)->method_string);
+
+				event_handler(&current_event_data);
+
+				/* Run next method on the list */
+				location_core_current_event_data_init(requested_method);
+				err = location_method_api_get(requested_method)->location_get(
+					&current_config.methods[current_method_index]);
+				return;
+			}
+			LOG_INF("LOCATION_REQ_MODE_ALL: all methods done");
+
+			/* Start from the beginning if in interval mode
+			 * for possible cancel within restarting interval.
+			 */
+			current_method_index = 0;
+		}
 	} else {
 		/* Do fallback to next preferred method */
```

Comment:
This comment is no longer accurate because of the "all" mode. The same goes also for the comment in the success branch above ("Location was acquired properly, finish location request"), because the request might not be yet be finished.

Reply:
finetuned